contracts/contracts/vault/OETHVaultAdmin.sol

```diff
@@ -1,12 +1,112 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { IStrategy } from "../interfaces/IStrategy.sol";
+import { IVault } from "../interfaces/IVault.sol";
 import { VaultAdmin } from "./VaultAdmin.sol";
 
 /**
  * @title OETH VaultAdmin Contract
  * @author Origin Protocol Inc
  */
 contract OETHVaultAdmin is VaultAdmin {
+    using SafeERC20 for IERC20;
+
+    address public immutable weth;
+
+    constructor(address _weth) {
+        weth = _weth;
+    }
+
+    /// @dev Simplified version of the deposit function as WETH is the only supported asset.
+    function _depositToStrategy(
+        address _strategyToAddress,
+        address[] calldata _assets,
+        uint256[] calldata _amounts
+    ) internal override {
+        require(
+            strategies[_strategyToAddress].isSupported,
+            "Invalid to Strategy"
+        );
+        require(
+            _assets.length == 1 && _amounts.length == 1 && _assets[0] == weth,
```

Comment:
Would be great if we can move `_amounts.length` to its own `require` statement so that we can know why it reverts when it does

Reply:
I'm good with the current line. If someone is only using WETH, everything on this line will be true.